chromium_src/chrome/browser/ui/views/frame/browser_frame_view_layout_linux.cc

```diff
@@ -3,14 +3,41 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "base/check_is_test.h"
 #include "brave/browser/ui/views/tabs/features.h"
+#include "chrome/browser/ui/views/frame/browser_view.h"
+#include "ui/views/window/caption_button_layout_constants.h"
 
 #include "src/chrome/browser/ui/views/frame/browser_frame_view_layout_linux.cc"
 
 int BrowserFrameViewLayoutLinux::NonClientTopHeight(bool restored) const {
-  if (tabs::features::ShouldShowVerticalTabs()) {
-    // Set minimum height to show caption buttons.
-    return 50;
+  if (!view_) {
+    CHECK_IS_TEST();
+    return OpaqueBrowserFrameViewLayout::NonClientTopHeight(restored);
+  }
+
+  if (view_->browser_view()->browser()->is_type_normal() &&
+      tabs::features::ShouldShowVerticalTabs()) {
+    if (!view_->ShouldShowCaptionButtons()) {
+      // In this case, the window manager might be forcibly providing system
+      // window title or it's in fullscreen mode. We shouldn't show title bar
+      // in this case.
+      return OpaqueBrowserFrameViewLayout::NonClientTopHeight(restored);
+    }
+
+    if (tabs::features::ShouldShowWindowTitleForVerticalTabs(
+            view_->browser_view()->browser())) {
+      return 30;
```

Comment:
Might be worth making a note about where this came from :smile:

Reply:
To be honest, it was just a magic number from me haha 😆 Now it's gone.